rotkehlchen/data_migrations/migrations/migration_11.py

```diff
@@ -42,10 +41,49 @@ def _get_optimism_transaction_fees(rotki: 'Rotkehlchen', progress_handler: 'Migr
         )
 
 
+def _get_optimism_transaction_fees_num_steps(rotki: 'Rotkehlchen') -> int:
+    """Returns the number of progress handler steps needed for the get optimism transaction fees"""
+    with rotki.data.db.conn.read_ctx() as cursor:
```

Comment:
This entire function makes no sense. You just duplicated `_get_optimism_transaction_fees` and are calling them both. Please don't do that. Duplicating functionality like this is unacceptable.

Just abstract it as a function.Use it at start. Return `txhash_to_id` and set its len to the steps and the continue with the logic that `_get_optimism_transaction_fees` had